### PR TITLE
chore: rename *.yaml to *.json

### PR DESCRIPTION
**Pull request by zirain**
cc @ShyunnY

### CODE REVIEW

File: internal/metrics/testdata/gauge_metric.json [hunk 1]
```diff
@@ -40,7 +40,7 @@
                   }
                 ],
                 "StartTime": "0001-01-01T00:00:00Z",
-                "Time": "2024-03-27T21:54:18.8562259+08:00",
+                "Time": "2024-03-31T11:25:29.204058+08:00",
```
Comment (Xunzhuo) [inline internal/metrics/testdata/gauge_metric.json:43]: will this update each time we run make generate ?

**Comment by ShyunnY** [comment]
> cc @ShyunnY

YAML is compatible with JSON. Do we need to modify it to JSON?

**Comment by zirain** [comment]
> > cc @ShyunnY
> 
> YAML is compatible with JSON. Do we need to modify it to JSON?

this will case yamllint tool panic

on the otherside, content should match with ext

**Comment by ShyunnY** [comment]
block

**Comment by zirain** [comment]
replaced by https://github.com/envoyproxy/gateway/pull/3077

**Pull request closed without merging**